pkg/tbtc/dkg.go

```diff
@@ -138,47 +147,40 @@ func (de *dkgExecutor) executeDkgIfEligible(
 // - Indexes of members selected to the signing group and controlled by this
 //   operator. The indexes are in range [1, `groupSize`]. The slice is nil if
 //   none of the selected signing group members is controlled by this operator.
-// - Addresses of all signing group members. There are always `groupSize`
-//   elements in this slice.
+// - Group selection result holding chain.OperatorID and chain.Address for
+//   operators selected to the signing group. There are always `groupSize`
+//   selected operators.
 func (de *dkgExecutor) checkEligibility(
 	dkgLogger log.StandardLogger,
-	seed *big.Int,
-) ([]uint8, chain.Addresses, error) {
-	selectedSigningGroupOperators, err := de.chain.SelectGroup(seed)
+) ([]uint8, *GroupSelectionResult, error) {
+	groupSelectionResult, err := de.chain.SelectGroup()
 	if err != nil {
 		return nil, nil, fmt.Errorf("selecting group not possible: [%v]", err)
 	}
 
-	dkgLogger.Infof("selected group members for DKG = %s", selectedSigningGroupOperators)
+	dkgLogger.Infof(
+		"selected group members for DKG = %s",
+		groupSelectionResult.OperatorsAddresses,
+	)
 
-	if len(selectedSigningGroupOperators) > de.chain.GetConfig().GroupSize {
+	if len(groupSelectionResult.OperatorsAddresses) > de.chain.GetConfig().GroupSize {
```

Comment:
> The requirements about the group size or threshold do not come from the chain

Is that true? Those are constants defined in `EcdsaDkg` and `EcdsaDkgValidator`. If we want to use different group sizes, we need to update the contracts and the client must adjust.

Reply:
What I mean is that we have a bunch of assumptions about the group size all over `pkg/tbtc`. For example, the `dkgAttemptMaximumProtocolBlocks` was chosen based on the tests of the 100-members group.

The chain has its own expectations about the group size but the client also has them, all across the board. In the current form, with the current constants, `pkg/tbtc` will work for groups of 100 but will not work for groups of 1000 on M1 dev machine. It will work for groups of 10 but retries may be highly inefficient. 

In other words, I think the group size of 100 is now also an expectation of `pkg/tbtc`.

Reply:
Ok, got it now. Thanks for the clarification! I'll change it.

Reply:
See 349bdfa8